src/routing_table/mod.rs

```diff
@@ -571,38 +538,29 @@ impl RoutingTable {
     /// more (i.e. only differ in one bit from our own prefix), they are removed and those contacts
     /// are returned. If the split is happening to our own section, our new prefix is returned in
     /// the optional field.
-    pub fn split(&mut self, ver_pfx: VersionedPrefix) -> (Vec<XorName>, Option<Prefix>) {
+    pub fn split(&mut self, prefix: Prefix) -> (Vec<XorName>, Option<Prefix>) {
         let mut result = vec![];
-        let (prefix, version) = ver_pfx.into();
+
         if prefix == self.our_prefix {
-            result = self.split_our_section(version);
+            result = self.split_our_section(prefix.version());
             return (result, Some(self.our_prefix));
         }
 
-        let (_version, to_split) = match self.sections.entry(prefix) {
-            Entry::Vacant(_) => return (result, None),
-            Entry::Occupied(ref entry) if entry.get().0 != version => {
-                debug!(
-                    "{:?} Not splitting section with {:?} ver. {}, \
-                     update is for a different version: {}",
-                    self.our_name,
-                    prefix,
-                    entry.get().0,
-                    version
-                );
-                return (result, None);
-            }
-            Entry::Occupied(entry) => entry.remove(),
+        let to_split = if let Some(section) = self.sections.remove(&prefix) {
```

Comment:
why now we can carry out remove directly without a version check ?

Reply:
The version check is implicit, because version is now part of `Prefix` and the equality comparison of `Prefix` involves version too.